### PR TITLE
[EDE-623] Switch xqueue_watcher from nose to pytest

### PR DESCRIPTION
**Pull request by tehreem-sadat**
**Description:**
-  Replace nose with pytest

**Packages removed**
nose

**New packages**
pytest-cov

### CODE REVIEW

File: .pep8 [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[pep8]
```
Comment (ziafazal) [inline .pep8]: We don't need to add `pep8`

File: requirements/test.in [hunk 2]
```diff
@@ -5,4 +5,9 @@
 -r production.txt
 
 mock
-nose
+
+pytest
+
+pytest-cov
```
Comment (ziafazal) [inline requirements/test.in:8]: no line breaks
no need to add `pytest` it already dependency of `pytest-cov`
no need to add `pep8`

File: Makefile [hunk 3]
```diff
@@ -29,7 +30,10 @@ travis.requirements:
 	pip install -q -r requirements/travis.txt --exists-action w
 
 test: test.requirements
-	nosetests --with-coverage --cover-package=xqueue_watcher --cover-xml
+	python -m pytest --cov=xqueue_watcher tests/
```
Comment (ziafazal) [inline Makefile]: i think command should be
`pytest --cov= xqueue_watcher --cov-report=xml`

**Review by ziafazal — CHANGES_REQUESTED** [review]
@tehreem-sadat Thanks for pulling this up.
We should avoid any python code changes purpose of this task is to only replace `nose` with `pytest`.
When you create upstream PR please use [BD-18] prefix to PR title instead of EDE-* .

File: requirements/test.in (hunk 2 above)
Comment (tehreem-sadat) [reply requirements/test.in:8]: done.

File: .pep8 (hunk 1 above)
Comment (tehreem-sadat) [reply .pep8]: done.

**Comment by tehreem-sadat** [comment]
@ziafazal all changes have been made. Kindly review it again.

**Review by ziafazal — APPROVED** [review]
@tehreem-sadat thanks for quick turn around could you please create upstream PR?

**Comment by tehreem-sadat** [comment]
Upstream PR link: https://github.com/edx/xqueue-watcher/pull/49

**Pull request merged**